program/rust/src/rust_oracle.rs

```diff
@@ -38,41 +45,119 @@ use crate::c_oracle_header::{
     PC_MAX_NUM_DECIMALS,
     PC_PROD_ACC_SIZE,
     PC_PTYPE_UNKNOWN,
+    PC_VERSION,
+    SUCCESSFULLY_UPDATED_AGGREGATE,
 };
 use crate::deserialize::{
     load,
     load_account_as,
     load_account_as_mut,
 };
 use crate::error::OracleResult;
-use crate::utils::pyth_assert;
 use crate::OracleError;
 
+use crate::utils::pyth_assert;
+
 use super::c_entrypoint_wrapper;
+const PRICE_T_SIZE: usize = size_of::<pc_price_t>();
+const PRICE_ACCOUNT_SIZE: usize = size_of::<PriceAccountWrapper>();
+
 
 ///Calls the c oracle update_price, and updates the Time Machine if needed
 pub fn update_price(
     _program_id: &Pubkey,
-    _accounts: &[AccountInfo],
+    accounts: &[AccountInfo],
     _instruction_data: &[u8],
     input: *mut u8,
 ) -> OracleResult {
-    //For now, we did not change the behavior of this. this is just to show the proposed structure
-    // of the program
-    c_entrypoint_wrapper(input)
+    let [_funding_account_info, price_account_info, _sysvar_clock] = match accounts {
+        [x, y, z] => Ok([x, y, z]),
+        _ => Err(ProgramError::InvalidArgument),
+    }?;
```

Comment:
I think you need to check that's a price account here : 
I recommended using `load_checked<pc_price_t>`